### PR TITLE
Editor fixes for Anki 2.1.52+

### LINKED ISSUE #92: Not working on 2.1.50
I imagine this is your own fault for changing the editor :P

Do you have any timeline on updating this?

### PR DESCRIPTION
**Pull request by kleinerpirat**
Closes #92.
Like you suggested earlier this week, I built this PR based on your 2.1.52 branch.

The following things are fixed in addition to your changes:
- Image occlusions
  - CSS is now injected with the EditorField API
  - because field content is retrieved asynchronously, I moved some of the Python logic to JS to prevent unnecessary back and forth messaging
- Wrapping with default cloze shortcuts

The PR is a crude hotfix, but it already suggests a move into a more JS based approach. Please let me know if I forgot anything important that needs fixing.

### CODE REVIEW

File: anki/web/editor.js [hunk 1]
```diff
@@ -119,8 +120,11 @@ var EditorCloset = {
     setupOcclusionEditor: async (closet, maxOcclusions) => {
         const fieldElements = [];
 
-        for (const field of instances[0].fields) {
-            fieldElements.push(await get(field.editingArea.editingInputs).find((input) => input.name === "rich-text").element);
+        for (const field of EditorField.instances) {
+            const richTextInputAPI = get(field.editingArea.editingInputs)
+                .find((input) => input.name === "rich-text");
+            subscriptionCallbacks.push(richTextInputAPI.preventResubscription());
```
Comment (kleinerpirat) [inline anki/web/editor.js:133]: This line was meant to be included in https://github.com/hgiesel/closet/pull/95/commits/a0f52a95e7099407c0d4c5299cff8fd14789c7df.

File: anki/web/editor.js [hunk 2]
```diff
@@ -73,6 +73,22 @@ var EditorCloset = {
...
+        return richTextEditable.innerHTML;
+    },
+
+    setFieldHTML: async (index, html) => {
+        const richTextEditable = await EditorCloset.getRichTextEditable(index);
+        richTextEditable.innerHTML = html;
+    },
+
+    getRichTextEditable: async (index) => {
+        return await get(EditorField.instances[index].editingArea.editingInputs)
```
Comment (hgiesel) [inline anki/web/editor.js:121]: `return await` is basically a noop

File: anki/web/editor.js [hunk 3]
```diff
@@ -3,35 +3,88 @@ var rushInOut = (x) => {
...
+    })();
+    
+    if (number_of_subs > 0) {
+        return subbed;
+    } else if (["", "<br>"].includes(old_html)) {
+        return replacement;
+    } else {
+        return `${replacement}<br>${old_html}`;
+    } 
+}
```
Comment (hgiesel) [inline anki/web/editor.js:33]: Why the Python-like formatting?

**Review by hgiesel — APPROVED** [review]
Awesome job! Thanks so much!
The comments are just minor things.

**Comment by hgiesel** [comment]
I've pushed it to Ankiweb, next are Custom styles and Asset manager :)

File: anki/web/editor.js (hunk 3 above)
Comment (kleinerpirat) [reply anki/web/editor.js:33]: I wanted to make it easier for you to spot the parts that were originally in Python - but with GitLens you'd see the commit message anyway, so there's no valid reason to keep it :D

Comment (kleinerpirat) [reply anki/web/editor.js:33]: I wanted to make it easier for you to spot the parts that were originally in Python - but with GitLens you'd see the commit message anyway, so there's no valid reason to keep it :D

**Pull request merged**